torch/csrc/generic/serialization.cpp

```diff
@@ -6,21 +6,21 @@
 
 void THPTensor_(writeMetadataRaw)(THTensor *self, int fd)
 {
-  SYSCHECK(write(fd, &self->nDimension, sizeof(long)));
-  SYSCHECK(write(fd, self->size, sizeof(long) * self->nDimension));
-  SYSCHECK(write(fd, self->stride, sizeof(long) * self->nDimension));
-  SYSCHECK(write(fd, &self->storageOffset, sizeof(long)));
+  SYSCHECK(write(fd, &self->nDimension, sizeof(self->nDimension)));
```

Comment:
Hum... Was this a bug previously? `nDimension` was always `int`, but we were reading/writing as `long`.
Won't this change make reading files backward-incompatible?